gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/kafka/client/GobblinKafkaConsumerClient.java

```diff
@@ -169,6 +169,8 @@ default long committed(KafkaPartition partition) {
 
   public default void assignAndSeek(List<KafkaPartition> topicPartitions, Map<KafkaPartition, LongWatermark> topicWatermarksMap) { return; }
 
+  public default void initializeClient(String topic) { return; }
```

Comment:
What's this method for?

Reply:
This method is what will contain the partition assignment and offset setting logic to initialize the consumer. Since `SpecChangeMonitor` and `HighLevelConsumer` expect the consumer to be a class with the interface of `GobblinKafkaConsumerClient` there needs to be a method in this interface that can be called to setup the client.